lib/memjs/memjs.js

```diff
@@ -84,21 +101,20 @@ Client.create = function(serversStr, options) {
 
 // Chooses the server to talk to by hashing the given key.
 Client.prototype.server = function(key) {
-  // TODO(alevy): should use consistent hashing and/or allow swapping hashing
-  // mechanisms
-  var total = this.servers.length;
-  var origIdx = (total > 1) ? (hashCode(key) % total) : 0;
-  var idx = origIdx;
-  var serv = this.servers[idx];
-  while (serv.wakeupAt &&
-      serv.wakeupAt > Date.now()) {
-    idx = (idx + 1) % total;
-    if (idx === origIdx) {
-      return null;
-    }
-    serv = this.servers[idx];
+  return this.getServerByIndex(this.serverIndex(key));
+};
+
+Client.prototype.serverIndex = function(key) {
+  var total = Object.keys(this.serverMap).length;
+  if (total === 1) {
+    return Object.keys(this.serverMap)[0];
```

Comment:
is `this.serverMap` immutable? Can we store `this._serverMapKeys` so we don't need to allocate the array twice per request?

Reply:
It seems like you should just be doing `return 0` here. Instead you are returning a server hostname string.

Reply:
yes, immutable, done.